lib/remap-tests/tests/assignment_fallible_for_infallible_expression.vrl

```diff
@@ -0,0 +1,15 @@
+# object: { "bar": "bar" }
+# result:
+#
+# error: unneeded error assignment
+#   ┌─ :2:1
+#   │
+# 2 │ ok, err = 5;
+#   │ ^^^^^^^   - because this expression cannot fail
+#   │ │          
+#   │ this error assignment is unneeded
+#   │
+#   = hint: assign to "ok", without assigning to "err"
+#   = see language documentation at: https://vector.dev/docs/reference/vrl/
+
+ok, err = 5;
```

Comment:
When reviewing, consider these early prototype messages.

The actual URLs, notes, and exact messages will be iterated on in future PRs. This PR's main purpose is to have the plumbing in place to allow us to improve these messages going forward.